src/providers/pkcs11_provider/key_management.rs

```diff
@@ -276,16 +277,17 @@ impl Pkcs11Provider {
 
         let mut template: Vec<CK_ATTRIBUTE> = Vec::new();
 
-        let public_key: RsaPublicKey = picky_asn1_der::from_bytes(&op.data).or_else(|e| {
-            format_error!("Failed to parse RsaPublicKey data", e);
-            remove_key_id(
-                &key_triple,
-                key_id,
-                &mut *store_handle,
-                &mut local_ids_handle,
-            )?;
-            Err(ResponseStatus::PsaErrorInvalidArgument)
-        })?;
+        let public_key: RsaPublicKey = picky_asn1_der::from_bytes(op.data.expose_secret())
```

Comment:
That is very interesting because `Secret` now allows use to see more precisely where we share our assets with dependencies. Currently, we clearly state in our TM that "depencencies are not checked for vulnerabilities". If we want to check that at some point, we can start by grepping whenever we use `expose_secret`. For example here, what do we know what `picky_asn1_der::from_bytes` is doing with our asset?